src/lib.rs

```diff
@@ -11,18 +13,113 @@ pub(crate) use visitor::Visitor;
 
 /// Parse the source code in `src_file` and return a `syn::File` that has all modules
 /// recursively inlined.
+///
+/// This is equivalent to using an `InlinerBuilder` with the default settings.
 pub fn parse_and_inline_modules(src_file: &std::path::Path) -> syn::File {
-    Visitor::<FsResolver>::new(src_file).visit()
+    InlinerBuilder::default().parse_and_inline_modules(src_file).unwrap()
+}
+
+/// A builder that can configure how to inline modules.
+///
+/// After creating a builder, set configuration options using the methods
+/// taking `&mut self`, then parse and inline one or more files using
+/// `parse_and_inline_modules`.
+#[derive(Debug)]
+pub struct InlinerBuilder {
+    root: bool,
+    error_not_found: bool,
+}
+
+/// A source location
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SourceLocation {
+    pub path: PathBuf,
+    pub line: usize,
+    pub column: usize,
+    _private: ()
+}
+
+impl SourceLocation {
+    pub(crate) fn new(path: &Path, span: proc_macro2::Span) -> Self {
+        SourceLocation {
+            path: path.into(),
+            line: span.start().line,
+            column: span.start().column,
+            _private: (),
+        }
+    }
+}
+
+/// An error that was encountered while inlining modules
+#[derive(Debug)]
+pub enum Error {
+    /// The contents for one or more modules could not be found
+    NotFound(Vec<(String, SourceLocation)>),
+    #[doc(hidden)]
+    __NonExhaustive,
+}
+
+impl Default for InlinerBuilder {
```

Comment:
Minor nit, but my understanding is that it's standard to put impls directly after the struct declaration. Do you mind moving these up?

Reply:
Done